src/webapp/context-providers/CurrentModuleContextProvider.tsx

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState, useMemo, useCallback } from "react";
+import { ModuleAccess } from "../../domain/entities/User";
+import { useHistory, useLocation } from "react-router-dom";
+import { defaultModuleContextState, CurrentModuleContext } from "../contexts/current-module-context";
+
+import { useAppContext } from "../contexts/app-context";
+
+export const CurrentModuleContextProvider: React.FC = ({ children }) => {
+    const history = useHistory();
+    const location = useLocation();
+    const queryParameters = useMemo(() => new URLSearchParams(location.search), [location]);
+    const moduleQueryParam = queryParameters.get("module");
+    const { currentUser } = useAppContext();
+
+    const [currentModuleAccess, setCurrentModuleAccess] = useState<ModuleAccess>(
+        defaultModuleContextState.currentModuleAccess
+    );
+
+    const [userModulesAccess, setUserModulesAccess] = useState<ModuleAccess[] | null>(null);
```

Comment:
[very minor] Unless specific reasons, we typically use the default empty value of useState, `undefined` instead of `null`

Reply:
Addressed in https://github.com/EyeSeeTea/glass-dev/commit/c664c49dc8abcbeb0baeceac61d29b7d7275cb08